File: backend/app/serializers.py
from bson import ObjectId
from fastapi import HTTPException

def object_id(value: str) -> ObjectId:
    try: return ObjectId(value)
    except Exception as exc: raise HTTPException(400, "Invalid ID") from exc

def article_response(article: dict, full: bool = False) -> dict:
    data = {"id": str(article["_id"]), "title": article["title"], "slug": article["slug"], "excerpt": article["excerpt"], "category": article["category"], "imageUrl": article.get("image_url"), "status": article["status"], "featured": article.get("featured", False), "author": article.get("author_name", "न्यूज़ डेस्क"), "publishedAt": article.get("published_at").isoformat() if article.get("published_at") else None, "updatedAt": article["updated_at"].isoformat(), "seoTitle": article.get("seo_title"), "seoDescription": article.get("seo_description"), "seoKeywords": article.get("seo_keywords"), "seoImageUrl": article.get("seo_image_url")}
    if full: data["body"] = article["body"]
    return data

File: backend/src/serializers.ts
import type {WithId,Document} from "mongodb";
import {iso} from "./utils.js";

export function articleResponse(article:WithId<Document>,full=false){const data:Record<string,unknown>={id:String(article._id),title:article.title,slug:article.slug,excerpt:article.excerpt,category:article.category,imageUrl:article.image_url,status:article.status,featured:article.featured??false,author:article.author_name??"न्यूज़ डेस्क",publishedAt:iso(article.published_at),updatedAt:iso(article.updated_at),seoTitle:article.seo_title,seoDescription:article.seo_description,seoKeywords:article.seo_keywords,seoImageUrl:article.seo_image_url};if(full)data.body=article.body;return data}
export function categoryResponse(item:WithId<Document>){return{id:String(item._id),name:item.name,slug:item.slug,parentId:item.parent_id?String(item.parent_id):null,active:item.active??true,position:item.position??0}}
export function audioResponse(item:WithId<Document>){return{id:String(item._id),title:item.title,filename:item.filename,size:item.size,active:item.active,position:item.position,audioUrl:`/audio/${item._id}/stream`}}
